drivers/gpio/gpio_keys_zephyr.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio_keys.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(zephyr_gpio_keys, LOG_LEVEL_DBG);
+
+#define DT_DRV_COMPAT zephyr_gpio_keys
+
+#define GPIO_KEYS_CFG_DEF(i)                                                                       \
+	{                                                                                          \
+		.spec = GPIO_DT_SPEC_GET(DT_DRV_INST(i), gpios),                                   \
+		.debounce_interval_ms = DT_INST_PROP(i, debounce_interval_ms),                     \
+	}
+
+struct gpio_keys_data {
+	const struct device *dev;
+	gpio_callback_handler_t callback;
+	struct gpio_callback cb_data;
+	struct k_work_delayable work;
+	uint32_t debounce_interval_ms;
+	int pin_state;
+	int is_stable;
+};
+
+/**
+ * Handle debounced gpio pin state.
+ */
+static void gpio_keys_change_deferred(struct k_work *work)
+{
+	struct gpio_keys_data *data = CONTAINER_OF(work, struct gpio_keys_data, work);
+	const struct device *dev = data->dev;
+	const struct gpio_keys_config *cfg = (struct gpio_keys_config *)dev->config;
+
+	const int new_pressed = gpio_keys_get_pin_raw(dev);
```

Comment:
This is getting data for the whole port, guess you may want to add a `& BIT(cfg->spec.pin)`.

Reply:
This looks to be already handled in `gpio_pin_get_raw`
https://github.com/zephyrproject-rtos/zephyr/blob/016f0d4fbe93dfe6a5b51ac565faf17981bf29ab/include/zephyr/drivers/gpio.h#L1188

Reply:
Oh my bad, thought this was returning the whole port.